### PR TITLE
Add rocm image to circleci docker builder

### PR DESCRIPTION
**Pull request by jeffdaily**
CC @ezyang @sunway513

### CODE REVIEW

**Comment by jeffdaily** [comment]
@ezyang first time we're trying to build a circleci rocm image.  I anticipate some rough edges and learning opportunities.  Following the instructions you sent to me earlier: https://github.com/pytorch/pytorch/wiki/Docker-image-build-on-CircleCI.

**Comment by seemethere** [comment]
Yea looks like the failure itself is related to mismatched clang version, you may also need to add your image here:

https://github.com/pytorch/pytorch/blob/master/.circleci/docker/build.sh#L39

**Comment by jeffdaily** [comment]
Image is already there.   Hasn't been attempted until now. 
 https://github.com/pytorch/pytorch/blob/master/.circleci/docker/build.sh#L185

Will look into clang mismatch, though it seems mismatch is because HCC compiler in ROCm environment is clang in this case.

File: .circleci/docker/build.sh [hunk 1]
```diff
@@ -295,7 +295,11 @@ if [ -n "$CLANG_VERSION" ]; then
   if !(drun clang --version 2>&1 | grep -qF "clang version $CLANG_VERSION"); then
     echo "CLANG_VERSION=$CLANG_VERSION, but:"
     drun clang --version
-    exit 1
+    if [[ "$image" == *-rocm* ]]; then
+      echo "For ROCm builds, version mismatch expected."
```
Comment (ezyang) [inline .circleci/docker/build.sh:299]: yuck

**Comment by ezyang** [comment]
I created the missing ECR repo, try again

File: .circleci/docker/build.sh (hunk 1 above)
Comment (seemethere) [reply .circleci/docker/build.sh:299]: Should this check just do something else in the case of rocm?

Like I'd expect it to check against some version or don't do the check at all.

Also, if this check is invalid, should we even do a `CLANG_VERSION` at all?

Comment (ezyang) [reply .circleci/docker/build.sh:299]: If you're not going to use clang, just undefine `CLANG_VERSION` in the config

Comment (jeffdaily) [reply .circleci/docker/build.sh:299]: Should we leave the ECR entry for  `pytorch-linux-xenial-rocm-py3.6-clang7` or create a new one without the clang7, `pytorch-linux-xenial-rocm-py3.6`?

**Comment by jeffdaily** [comment]
@ezyang @seemethere Should we leave the ECR repo for `pytorch-linux-xenial-rocm-py3.6-clang7` or create a new one without the clang7, `pytorch-linux-xenial-rocm-py3.6`?

File: .circleci/docker/build.sh (hunk 1 above)
Comment (jeffdaily) [reply .circleci/docker/build.sh:299]: Discussed offline.  New ECR entry created for later.  For now, keep everything as-is (e.g., installing clang7 even though it isn't used) to expedite the process of getting ROCm CI switched to the new image.

A follow up PR will remove the clang7 install and rocm-specific work-around in the install script.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**